### PR TITLE
Make logger more compatible

### PR DESCRIPTION
**Pull request by shadowwalker**
Use `str.format` to complete format string internally, this helps `tenacity` to be more compatible with other loggers such as `loguru`.

Without this PR the logging shows `tenacity.before_sleep:log_it:41 - Retrying %s in %s seconds as it %s %s.` when passing in logger from loguru.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,5 +8,6 @@ AUTHORS
 ChangeLog
 .eggs/
 doc/_build
+.venv/
```
Comment (jd) [inline .gitignore:11]:
```suggestion
```

File: tenacity/__init__.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def retry(fn):
 @t.overload
 def retry(*dargs, **dkw):  # noqa
     # type: (...) -> t.Callable[[WrappedFn], WrappedFn]
-    """Type signature for the @retry() decorator constructor."""
+    """Type signature for the @retry() decorator constructor."""  # noqa: D402
```
Comment (jd) [inline tenacity/__init__.py:105]: Sounds unrelated?

**Review by jd — DISMISSED** [review]
Not a fan of this. Loguru not implementing the standard logging interface is their choice but making it paid by every library out there is not an option.

Since in that case, it's not too hard, I'd be ok with this. I'm just worried it'll be broken since there's no test and no easy way to add tests.

File: .gitignore (hunk 1 above)
Comment (shadowwalker) [reply .gitignore]: Updated

File: tenacity/__init__.py (hunk 2 above)
Comment (shadowwalker) [reply tenacity/__init__.py]: Updated

**Comment by jd** [comment]
@mergifyio update

**Review by jd — APPROVED** [review]

**Pull request merged**